src/Indicator.vala

```diff
@@ -338,6 +342,74 @@ public class Sound.Indicator : Wingpanel.Indicator {
         return main_grid;
     }
 
+    /* Handles both SMOOTH and non-SMOOTH events (although is only sent SMOOTH normally).
+     * In order to deliver smooth volume changes it:
+     * * ignores whether "natural scrolling" is set since this does not make sense in this context.
+     * * accumulates very small changes until they become significant.
+     * * ignores rapid changes in direction.
+     * * responds to both horizontal and vertical scrolling.
+     * In the case of diagonal scrolling, it takes the larger of the two directions.
+     */
+    private bool handle_scroll_event (Gdk.EventScroll e, out double dir) {
+        dir = 0.0;
+        switch (e.direction) {
+            case Gdk.ScrollDirection.SMOOTH:
+                if (same_sign (e.delta_x, total_x_delta)) {
+                    total_x_delta += e.delta_x;
+                } else {
+                    total_x_delta = 0;
+                }
+
+                if (same_sign (e.delta_y, total_y_delta)) {
+                    total_y_delta += e.delta_y;
+                } else {
+                    total_y_delta = 0.0;
+                }
+                break;
+
+            case Gdk.ScrollDirection.UP:
+                total_y_delta = -1.0;
+                total_x_delta = 0.0;
+                break;
+            case Gdk.ScrollDirection.DOWN:
+                total_y_delta = 1.0;
+                total_x_delta = 0.0;
+                break;
+            case Gdk.ScrollDirection.LEFT:
+                total_x_delta = -1.0;
+                total_y_delta = 0.0;
+                break;
+            case Gdk.ScrollDirection.RIGHT:
+                total_x_delta = 1.0;
+                total_y_delta = 0.0;
+                break;
+            default:
+                break;
+        }
+
+        dir = total_x_delta.abs () > total_y_delta.abs () ? total_x_delta : total_y_delta;
+
+        /* DO not honor natural scroll setting */
+        if (natural_scroll) {
+            dir *= -1.0;
+        }
+
```

Comment:
Is it just me or should we reverse the vertical direction? For me with these changes, regardless of natural scroll, when I move my finger up the volume goes down. Which seems backward to me personally. 
The horizontal scrolling seems consistent with the Scale movement.

Reply:
Not sure.  Best get UX team opinion on this.

Reply:
@elementary/ux ?